src/pages/ExecutionsPage.jsx

```diff
@@ -1,89 +1,89 @@
+  MODULE_EXECUTION_START_DATE,
+  TRANSFORMATION,
+} from "../constants/vocabulary.js";
 import Rest from "../rest/Rest.jsx";
 import { Col, Container, Row, Table } from "react-bootstrap";
 import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
 import { faMugHot, faEdit, faQuestion } from "@fortawesome/free-solid-svg-icons";
 import { Link } from "react-router-dom";
+import dayjs from "dayjs";
+import Spinner from "../components/spinner/Spinner.jsx";
 
-class ExecutionsPage extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      data: [],
-    };
-  }
+const ExecutionsPage = () => {
+  const [data, setData] = useState([]);
+  const [loading, setLoading] = useState(true);
 
-  componentDidMount() {
+  useEffect(() => {
     Rest.getExecutions().then((response) => {
       console.log(response);
-      this.setState({ data: response });
+      setData(response);
+      setLoading(false);
     });
+  }, []);
+
+  if (loading) {
+    return <Spinner />;
   }
 
-  render() {
-    if (this.state.data === []) {
-      return <h3>Loading</h3>;
-    } else {
-      return (
-        <>
-          <h3>Executions</h3>
-          <Table striped bordered hover>
-            <thead>
-              <tr>
-                <th>Status</th>
-                <th>Name</th>
-                <th>Started</th>
-                <th>Finished</th>
-                <th>Duration</th>
-                <th style={{ textAlign: "center" }}>Action</th>
+  return (
+    <>
+      <h3 className="mt-3">Executions</h3>
+      <Table striped bordered hover>
+        <thead>
+          <tr>
+            <th>Status</th>
+            <th>Name</th>
+            <th>Started</th>
+            <th>Finished</th>
+            <th>Duration</th>
+            <th style={{ textAlign: "center" }}>Action</th>
+          </tr>
+        </thead>
+        <tbody>
+          {data
+            .filter((v) => v !== null)
+            .map((data, key) => (
+              <tr key={key}>
+                <td align={"center"}>
+                  <FontAwesomeIcon icon={faMugHot} />
+                </td>
+                <td>{data[DISPLAY_NAME]}</td>
+                <td>{dayjs(data[MODULE_EXECUTION_START_DATE]).format("HH:mm:ss")}</td>
+                <td>{dayjs(data[MODULE_EXECUTION_FINISH_DATE]).format("HH:mm:ss")}</td>
+                <td>{data[EXECUTION_DURATION]}ms</td>
+                <td>
+                  <Container>
+                    <Row>
+                      <Col>
+                        <Link to={`/script?file=${data[ABSOLUTE_PATH]}&transformation=${data[TRANSFORMATION]}`}>
+                          <FontAwesomeIcon icon={faEdit} />
+                        </Link>
+                      </Col>
+                      <Col
+                        onClick={() => {
+                          window.open(
+                            data["http://onto.fel.cvut.cz/ontologies/s-pipes/rdf4j-transformation-id"],
```

Comment:
Why we do not have this as constant somewhere?

Reply:
done